frontend/apps/marketing/src/components/editorialCard/EditorialCard.tsx

```diff
@@ -0,0 +1,95 @@
+import {EntryFields} from 'contentful';
+import {useMemo} from 'react';
+
+import EditorialCard, {
+  EDITORIAL_CARD_LAYOUTS,
+} from '@code-dot-org/component-library/cms/editorialCard';
+
+import {fontAwesomeV6BrandIconsMap} from '@/components/common/constants';
+import {LinkEntry} from '@/types/contentful/entries/Link';
+import {ExperienceAsset} from '@/types/contentful/ExperienceAsset';
+
+export enum EDITORIAL_CARD_CONTENTFUL_LAYOUTS {
+  HORIZONTAL_WITH_IMAGE = 'horizontal_with_image',
+  VERTICAL_WITH_IMAGE = 'vertical_with_image',
+  VERTICAL_WITH_ICON = 'vertical_with_icon',
+}
+
+export interface EditorialCardContentfulProps {
+  layoutOpt: EDITORIAL_CARD_CONTENTFUL_LAYOUTS;
+  heading: EntryFields.Text;
+  text: EntryFields.Text;
+  iconName?: EntryFields.Text;
+  image?: ExperienceAsset | string;
+  linkEntry?: LinkEntry;
+}
+
+const EditorialCardContentful: React.FC<EditorialCardContentfulProps> = ({
+  layoutOpt,
+  heading,
+  text,
+  image,
+  iconName,
+  linkEntry,
+}) => {
+  const layout = useMemo(() => {
+    switch (layoutOpt) {
+      case EDITORIAL_CARD_CONTENTFUL_LAYOUTS.HORIZONTAL_WITH_IMAGE:
+        return EDITORIAL_CARD_LAYOUTS.HORIZONTAL;
+      case EDITORIAL_CARD_CONTENTFUL_LAYOUTS.VERTICAL_WITH_IMAGE:
+      case EDITORIAL_CARD_CONTENTFUL_LAYOUTS.VERTICAL_WITH_ICON:
+        return EDITORIAL_CARD_LAYOUTS.VERTICAL;
+    }
+  }, [layoutOpt]);
+
+  const media = useMemo(() => {
+    switch (layoutOpt) {
+      case EDITORIAL_CARD_CONTENTFUL_LAYOUTS.HORIZONTAL_WITH_IMAGE:
+      case EDITORIAL_CARD_CONTENTFUL_LAYOUTS.VERTICAL_WITH_IMAGE:
+        return image && {src: image as string};
+      case EDITORIAL_CARD_CONTENTFUL_LAYOUTS.VERTICAL_WITH_ICON:
+        return (
+          iconName && {
+            iconName,
+            iconStyle: 'solid' as const,
```

Comment:
Rather than casting, we can ensure the correct type from the start, see above suggestion